File: src/test/java/com/gitlab/lae/intellij/actions/tree/PopupActionTest.java
package com.gitlab.lae.intellij.actions.tree;

import com.intellij.ide.DataManager;
import com.intellij.ide.IdePopupManager;
import com.intellij.openapi.ui.popup.JBPopupFactory;
import org.junit.Test;

import static org.junit.Assert.assertTrue;
import static org.mockito.Mockito.mock;

public final class PopupActionTest {

    @Test
    public void enabledInModalContext() {
        assertTrue(newPopupAction().isEnabledInModalContext());
    }

    @Test
    public void isDumpAware() {
        assertTrue(newPopupAction().isDumbAware());
    }

    private PopupAction newPopupAction() {
        return new PopupAction(
                mock(ActionNode.class),
                new IdePopupManager(),
                mock(JBPopupFactory.class),
                mock(DataManager.class)
        );
    }
}

File: src/test/java/com/gitlab/lae/intellij/actions/tree/PopupActionTest.kt
package com.gitlab.lae.intellij.actions.tree

import com.intellij.ide.IdePopupManager
import com.nhaarman.mockitokotlin2.mock
import org.junit.Assert.assertTrue
import org.junit.Test

class PopupActionTest {

  @Test
  fun `enabled in modal context`() {
    assertTrue(newPopupAction().isEnabledInModalContext)
  }

  @Test
  fun `is dump aware`() {
    assertTrue(newPopupAction().isDumbAware)
  }

  private fun newPopupAction() =
    PopupAction(action(), mock(), IdePopupManager(), mock(), mock())
}
